### PR TITLE
Privacy Dashboard refactor

### PR DESCRIPTION
**Pull request by jaceklyp**
**Required**:

Task/Issue URL: https://app.asana.com/0/1201392122292466/1207677104594315/f
iOS PR: https://github.com/duckduckgo/iOS/pull/3038/
macOS PR: https://github.com/duckduckgo/macos-browser/pull/2943
What kind of version bump will this require?: Major

**Description**:
- remove old pixels,
- remove user behavior toast experiment,
- refactor Privacy Dashboard,
- add documentation,
- test Privacy Dashboard

**Steps to test this PR**:
1. Run tests
2. Report broken site from both dashboard/app menu; make sure it works.
3. Toggle protection on/off and see if it works.
4. On toggle protections off you should enter reporting flow, see if it works on send/do not send/dismiss.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Sources/PrivacyDashboard/PrivacyDashboardMode.swift [hunk 1]
```diff
@@ -16,13 +16,25 @@
 //  limitations under the License.
 //
 
-/// Type of web page displayed
-public enum PrivacyDashboardMode: Equatable {
-
+/// Represents the type of web page displayed within the privacy dashboard flow.
```
Comment (ayoy) [inline Sources/PrivacyDashboard/PrivacyDashboardMode.swift:19]: Thanks for taking time to add documentation. I think we could do it more inside BSK 👏

File: Sources/PrivacyDashboard/PrivacyDashboardURLBuilder.swift [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        guard let baseURL = Bundle.privacyDashboardURL else { fatalError() }
+        self.url = baseURL
+        self.configuration = configuration
+    }
+
+    func build() -> URL {
+        url = configuration.addScreenParameter(to: url)
+        url = configuration.addBreakageScreenParameterIfNeeded(to: url)
+        url = configuration.addCategoryParameterIfNeeded(to: url)
+        url = configuration.addOpenerParameterIfNeeded(to: url)
```
Comment (ayoy) [inline Sources/PrivacyDashboard/PrivacyDashboardURLBuilder.swift:90]: These functions could be rewritten as URL extension functions taking configuration parameter and then you could do something like this:
```swift
return url.addingScreenParameter(from: configuration)
    .addingBreakageScreenParameterIfNeeded(from: configuration)
    .addingCategoryParameterIfNeeded(from: configuration)
    .addingOpenerParameterIfNeeded(from: configuration)
```

But it really isn't crucial :)

**Review by ayoy — APPROVED** [review]
This looks great! I did some testing on iOS and macOS and found no regressions. The code looks clean and I don't really have improvement suggestions. Great job! 💪

File: Sources/PrivacyDashboard/PrivacyDashboardURLBuilder.swift (hunk 2 above)
Comment (jaceklyp) [reply Sources/PrivacyDashboard/PrivacyDashboardURLBuilder.swift]: I changed this :)

**Pull request merged**